src/factories.js

```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import components from './index';
+
+let factories = {};
+
+Object.keys(components).forEach(function (name) {
+  if (components[name] instanceof React.Component.constructor) {
```

Comment:
Those generated scripts can come in a follow up pull request or we can try and bundle with this one. You'll need to consider both a CommonJS and AMD generation :wink: